spyder/plugins/ipythonconsole/utils/kernelspec.py

```diff
@@ -84,41 +55,30 @@ def argv(self):
         if CONF.get('main_interpreter', 'default'):
             pyexec = get_python_executable()
         else:
-            # Avoid IPython adding the virtualenv on which Spyder is running
-            # to the kernel sys.path
-            os.environ.pop('VIRTUAL_ENV', None)
             pyexec = CONF.get('main_interpreter', 'executable')
             if not is_python_interpreter(pyexec):
                 pyexec = get_python_executable()
                 CONF.set('main_interpreter', 'executable', '')
                 CONF.set('main_interpreter', 'default', True)
                 CONF.set('main_interpreter', 'custom', False)
 
-        # Part of spyder-ide/spyder#11819
-        is_different = is_different_interpreter(pyexec)
-
         # Command used to start kernels
-        if is_different and is_conda_env(pyexec=pyexec):
```

Comment:
All iPython consoles should be treated with parity, i.e. it shouldn't matter whether they share the same interpreter location on disk as Spyder's runtime interpreter. In all cases, they should be launched with clean environments and independent of Spyder's runtime environment. Since ``is_different`` only tells me whether the new iPython console interpreter is the same as Spyder's runtime interpreter, it serves no purpose; the ``spyder_kernel.console`` must still be launched with the appropriate script and environment variables.

Reply:
> All iPython consoles should be treated with parity, i.e. it shouldn't matter whether they share the same interpreter location on disk as Spyder's runtime interpreter

This sounds good in principle but it introduces some headaches in practice that I would like to avoid. For instance, with this approach same-interpreter kernels won't restart automatically if they die while running your code.

So I'd prefer to maintain this difference, just to avoid the risk of a very disruptive change in the way we handle kernels in 4.2 (we had that in 4.1.0 and it was a nightmare of bug reports).